rmw_connext_shared_cpp/include/rmw_connext_shared_cpp/shared_functions.hpp

```diff
@@ -229,6 +229,13 @@ wait(const char * implementation_identifier,
           }
         }
         if (fixed) {
+          // Reset the fixed guard conditions to avoid being woken up
+          // immediately next time.
+          DDSGuardCondition * guard = static_cast<DDSGuardCondition *>((*attached_conditions)[i]);
+          retcode = guard->set_trigger_value(DDS_BOOLEAN_FALSE);
+          if (retcode != DDS_RETCODE_OK) {
+            RMW_SET_ERROR_MSG("failed to set trigger value");
```

Comment:
I thought about that, but this is in the destructor for a stack variable that's being torn down on the way out of the function that declares it. I don't have a way to return an error code, and returning early would potentially avoid other useful cleanup that's happening in the destructor.